src/analyzer/protocol/mysql/mysql-protocol.pac

```diff
@@ -240,35 +269,141 @@ type Handshake_Response_Packet = case $context.connection.get_version() of {
 
 type Handshake_Credentials_v10 = record {
 	username : NUL_String;
-	password : bytestring &restofdata;
+	password : LengthEncodedString;
+};
+
+type Connection_Attribute = record {
+	name  : LengthEncodedString;
+	value : LengthEncodedString;
+};
+
+type Handshake_Connection_Attributes = record {
+	length : uint8;
+	attrs  : Connection_Attribute[] &until($input.length() == 0);
+} &length = length+1;
+
+type Handshake_Plain_v10(cap_flags: uint32) = record {
+	credentials: Handshake_Credentials_v10;
+	have_db     : case ( cap_flags & CLIENT_CONNECT_WITH_DB ) of {
+		CLIENT_CONNECT_WITH_DB -> database: NUL_String;
+		0x0 -> none_1    : empty;
+	};
+	have_plugin : case ( cap_flags & CLIENT_PLUGIN_AUTH ) of {
+		CLIENT_PLUGIN_AUTH -> auth_plugin: EmptyOrNUL_String;
```

Comment:
I haven't quite figured out the `EmptyOrNUL_String` here. If I change it to `NUL_String` (which I believe would be right), `auth.trace` fails.

The one part that stands out is that in `auth.trace`: The server does not advertise `CLIENT_PLUGIN_AUTH` and I wonder if in `Handshake_Credentials_v10` we cannot expect `password` / `auth_response` data at all then (where we instead currently consume the 0 as string length).